src/App.tsx

```diff
@@ -9,41 +9,52 @@ import {
 import { WebView } from "react-native-webview";
 import { widget } from "./trustly";
 import { shouldOpenInAppBrowser } from "./oauth-utils";
-
 import { InAppBrowser } from 'react-native-inappbrowser-reborn'
-import {Colors} from 'react-native/Libraries/NewAppScreen';
+import { Colors } from 'react-native/Libraries/NewAppScreen';
+import { getRequestSignature } from "./utils/signature";
+import { EstablishData } from "./types";
 
 export default class App extends Component {
   trustlyWebView = null;
 
-  establishData = {
-    accessId: "<YOUR_ACCESS_ID>",
-    merchantId: "<YOUR_MERCHANT_ID>",
-    currency: "USD",
-    amount: "2.00",
-    merchantReference: "<unique reference code from your app>",
-    paymentType: "Retrieval",
-    returnUrl: "/returnUrl",
-    cancelUrl: "/cancelUrl",
-    customer: {
-      name: "John",
-      address: {
-        country: "US",
-      },
-    },
-    metadata:{
-      integrationContext: "InAppBrowserNotify",
-      urlScheme: "in-app-browser-rn://"
-    },
-    requestSignature: "HT5mVOqBXa8ZlvgX2USmPeLns5o=",
-  };
-
   state = {
     bounceValue: new Animated.Value(1000),
+    establishData: null
   };
 
   constructor(props) {
     super(props);
+
+    (async () => {
```

Comment:
Please move this IIFE to the `componentDidMount()` method instead of `constructor()`.

This will avoid calling `setState` when the component is not yet mounted.